src/encode-macros/src/derive_fixed_storable.rs

```diff
@@ -0,0 +1,128 @@
+use proc_macro::TokenStream;
+use quote::ToTokens;
+use syn::{parse, Data, DataStruct, DeriveInput, Field, Fields, Generics, Ident, Index, Type};
+
+pub fn derive_fixed_storable(tokens: TokenStream) -> TokenStream {
+    let ast: DeriveInput = parse(tokens).unwrap();
+
+    match ast.data {
+        Data::Struct(struct_data) => {
+            derive_fixed_storable_struct(&struct_data, &ast.ident, &ast.generics)
+        }
+        Data::Enum(_) => panic!("enums not supported"),
+        Data::Union(_) => panic!("unions not supported"),
+    }
+}
+
+fn derive_fixed_storable_struct(
+    struct_data: &DataStruct,
+    struct_name: &Ident,
+    generics: &Generics,
+) -> TokenStream {
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    let max_size_tokens = generate_max_size(struct_data);
+    let is_fixed_tokens = generate_is_fixed(struct_data);
+
+    let field_names = get_field_names(struct_data);
+    let field_types = get_field_types(struct_data);
+
+    let serialize_field_tokens = quote::quote! {
+        #(
+            result.extend_from_slice(&self.#field_names.to_bytes());
+        )*
+    };
+
+    // use universal names that wotk both for named fields and in a tuple
+    let local_names: Vec<_> = field_names
+        .iter()
+        .map(|id| quote::format_ident!("decoded_{}", id.to_string()))
+        .collect();
```

Comment:
We are doing quite a lot of iteration right here. I think we can optimize these since they are reusing almost all the fields but doing separate retrieval.  I have done the best attempt to try to optimize this. Check the suggestion.

Reply:
To say honestly I don't think that this optimization will affect somehow on build time (runtime is not affected at all) because it is hard to imagine hundreds of members in a single structure. But anyway, I removed several iterations in a bit different way